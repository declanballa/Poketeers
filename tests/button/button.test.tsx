import 'jsdom-global/register';
import Button from '@components/button/button';
import { shallow, ShallowWrapper } from 'enzyme';


const testOnClick = () => { /* Dummy function */ };

describe('when button component renders', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Button
        labelText={ 'Test'}
        onClick={ testOnClick }
        type={ 'grass' } />
    );
  });


  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should be possible to open menu with Spacebar', () => {
    //
  });
});

