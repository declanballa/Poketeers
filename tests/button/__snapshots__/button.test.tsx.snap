// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when button component renders should match snapshot 1`] = `
<div
  className="button button--grass"
  onClick={[MockFunction]}
>
  <span
    className="button--grass__label"
  >
    Test
  </span>
</div>
`;
